addons/hr_timesheet/static/src/js/task_with_hours.js

```diff
@@ -38,7 +38,9 @@ const TaskWithHours = FieldMany2One.extend({
      * @private
      */
     _renderEdit: function (){
-        this.m2o_value = this._getDisplayNameWithoutHours(this.m2o_value);
+        if (this.m2o_value) {
+            this.m2o_value = this._getDisplayNameWithoutHours(this.m2o_value);
+        }
```

Comment:
I add this because after enabling task creation from timer sometime it get undefined m2o_value [image](https://www.awesomescreenshot.com/image/21865994?key=26cf1d6db35d450d873b5c7da10bf776) that's I added this.

Reply:
I don't think it is that the problem because if `this.m2o_value = undefined` then `value && value.split('\u00A0')[0]` should return the value in `value` since we check `value` before using the `split` method. 🤔 
So, I don't think your changes fixed the issue. 🤔

Reply:
Done. it's an issue from the `_getDisplayName` method so I overwrite that method and check `value` before using the `split` method.